### ISSUE TITLE
Support multiple <Toaster> elements

### ISSUE THREAD

**Issue by mrmiguu** [timolins/react-hot-toast#158]
This might be outside of the scope for this project, but the ability to have multiple `<Toaster>` elements on a page (where `toast()` calls could have a `ToasterID` passed in) would be pretty cool and add a lot flexibility.

Example:
```tsx
toast('This appears as a toast relative to the <div> parent', relativeToasterId)
toast('This appears as a normal toast')

...

return (
  <>
    <div>
      <Toaster id={relativeToasterId} />
    </div>

    <Toaster />
  </>
)
```

**Comment by timolins** [comment]
Hi @mrmiguu! Great suggestion - considered this for v2, but decided to not to do it atm. 

Let's use this issue to see how the demand is. 👍

**Comment by oscar-at-seatti** [comment]
This would be super helpful.

**Comment by TrejoCode** [comment]
Please, consider, would be amazing feature

**Comment by mrmiguu** [comment]
https://github.com/timolins/react-hot-toast/pull/159
☝️☝️☝️

This PR, if merged, essentially works like this:
```tsx
toast('hello', { toasterId: 'unique-id-here' })

...

<Toaster toasterId="unique-id-here" />
```

This way, only _that Toaster_ will be able to receive `toast`s to-and-from the string `"unique-id-here"`.

**Comment by rickyzhangca** [comment]
looking forward to having this!

**Comment by thanhngvpt** [comment]
hi all, 
how about this

**Comment by mattellig** [comment]
This would be an extremely valuable feature! Please consider adding.

**Comment by Mirajjj** [comment]
Yeah. That is what we need right now. For now we will have to support a different functionality with the same behavior.

**Comment by Benimation** [comment]
My guess is that this doesn't use React Context yet.

If there could be something like a `<ToastProvider>`, you could use that multiple times and even nest them, where the innermost provider determines which one should be used. That could be really useful if you have one for general app messages, and another one inside of a sidebar or something like that.

I also noticed that it doesn't work out-of-the-box when you have a component library (in a different package) and want to toast things from there to the main app. A context could make that really easy, but for now I just passed the `toast()` function to our own context and retrieve it from there rather than importing it.

**Comment by Arcanorum** [comment]
pls ser?

**Comment by eugene-f02** [comment]
This would be an extremely valuable feature!

**Comment by sspenst** [comment]
+1, would definitely be helpful. Having to do some workarounds right now to get toasts to work with headless UI dialogs.

**Comment by codigoisaac** [comment]
Please consider this feature :)

**Comment by Trickkyyy** [comment]
This would be a great feature,hands down

**Comment by alexgoff** [comment]
would love to have this feature as well

**Comment by RyanTippsTX** [comment]
+1
I need some toasts to overlay on my video player

**Comment by AntoninoGargiuloBID** [comment]
Any news? It would be a great feature.

**Comment by r-josue** [comment]
Any news? It would be a great feature.

**Comment by emilywilder** [comment]
Just wanted to add my interest in this feature

**Comment by fccjr24** [comment]
need this too

**Comment by georgiosApo** [comment]
Any news on this? Been a while, would be a amazing feature to have 🤌

**Comment by fellypsantos** [comment]
I think the owner is not quite available to manage the repo, we have many pending pull requests, few months ago I was needing a feature, I sent a pull request and it's still pending, so I've forked the repo, implemented the feature that I was needing, and sent to NPM, temporary solution.

**Comment by timolins** [comment]
I started a PR (#383) for this! You can follow along there.
